File: mode-shift-discrete-choice-model/chi-square-analysis/analysis_04_trip_purpose_primary_mode/analysis.R
# ==============================================================================
# analysis.R - Analysis 04: Trip Purpose x Primary Mode
# ==============================================================================
# Author : Majharul Islam
# Date   : 2026-08-07
# R      : 4.3+
# ==============================================================================

SCRIPT_DIR <- tryCatch(
  dirname(rstudioapi::getSourceEditorContext()$path),
  error = function(e) getwd()
)
REPO_ROOT <- dirname(SCRIPT_DIR)
DATA_PATH <- file.path(REPO_ROOT, "data", "Mode_shift_bubt.xlsx")
OUTPUT_DIR <- SCRIPT_DIR

source(file.path(REPO_ROOT, "chi_square_utils.R"))

message(rep("=", 60))
message("ANALYSIS 04 (R): Trip Purpose x Primary Mode")
message(rep("=", 60))

results <- run_full_analysis(
  data_path        = DATA_PATH,
  row_keywords     = c("primary purpose"),
  col_keywords     = c("PRIMARY mode"),
  row_var_label    = "Trip Purpose x Primary Mode (Row)",
  col_var_label    = "Trip Purpose x Primary Mode (Col)",
  table_title      = "Table 4.5: Mode by Purpose",
  output_dir       = OUTPUT_DIR,
  row_order        = c("Education (Classes/Exam)", "Work / Job", "Personal Business"),
  col_order        = c("Public Bus", "MRT (Metro Rail)", "Personal Motorcycle", "Ridesharing (Uber/Pathao)"),
  use_monte_carlo_ffh         = TRUE,
  use_monte_carlo_permutation = FALSE,
  n_replications   = 99999,
  seed             = 42
)

cat("\n", rep("=", 60), "\n")
cat(paste0("ANALYSIS 04 RESULTS -- Trip Purpose x Primary Mode\n"))
cat(rep("=", 60), "\n")
cat(sprintf("  Sample size (N)      : %d\n", results$n_total))
cat(sprintf("  Chi-square           : chi2(%d) = %.4f\n", results$degrees_of_freedom, results$chi2_statistic))
cat(sprintf("  P-value              : %.6f\n", results$p_value))
cat(sprintf("  Cramer V             : %.4f (%s)\n", results$cramers_v, results$effect_size_label))
cat(sprintf("  Decision: %s\n", results$decision))
cat(rep("=", 60), "\n")
cat(sprintf("Outputs saved to: %s\n", OUTPUT_DIR))

File: mode-shift-discrete-choice-model/chi-square-analysis/analysis_09_primary_mode_dedicated_bus/analysis.py
#!/usr/bin/env python3
"""
Analysis 09: Primary Mode x Dedicated BUBT Bus
Author : Majharul Islam
Date   : 2026-08-07
"""

import logging
import sys
from pathlib import Path

SCRIPT_DIR = Path(__file__).resolve().parent
REPO_ROOT = SCRIPT_DIR.parent
DATA_PATH = REPO_ROOT / "data" / "Mode_shift_bubt.xlsx"
OUTPUT_DIR = SCRIPT_DIR

sys.path.insert(0, str(REPO_ROOT))
from chi_square_utils import run_full_analysis

logging.basicConfig(level=logging.INFO, format="%(asctime)s [%(levelname)s] %(name)s — %(message)s")


def main() -> None:
    results = run_full_analysis(
        data_path=str(DATA_PATH),
        row_keywords=["PRIMARY mode"],
        col_keywords=["dedicated BUBT"],
        row_var_label="Primary Mode x Dedicated BUBT Bus (Row)",
        col_var_label="Primary Mode x Dedicated BUBT Bus (Col)",
        table_title="Table 4.11: Willingness to Switch to Dedicated BUBT Bus (n, row %)",
        output_dir=str(OUTPUT_DIR),
        row_order=['Public Bus', 'MRT (Metro Rail)', 'Personal Motorcycle', 'Ridesharing (Uber/Pathao)'],
        col_order=['Yes', 'No'],
        use_monte_carlo_ffh=False,
        use_monte_carlo_permutation=False,
        row_recode=None,
    )
    print(f"Analysis 09 Complete: chi2 = {results['chi2_statistic']:.4f}, p = {results['p_value']:.4f}")


if __name__ == "__main__":
    main()
